src/Core/Settings.h

```diff
@@ -974,6 +974,8 @@ class IColumn;
     M(String, bool_true_representation, "true", "Text to represent bool value in TSV/CSV formats.", 0) \
     M(String, bool_false_representation, "false", "Text to represent bool value in TSV/CSV formats.", 0) \
     \
+    M(Bool, input_format_numbers_enum_on_conversion_error, false, "Throw an exception during Numbers to Enum conversion if the value does not exist in Enum.", 0) \
```

Comment:
As I can see, this setting is used only in cast, not in input formats. Let's move it in common settings, remove `input_format` prefix and remove it from `FormatSettings`.
Or this settings should be also used in some input formats to validate number -> enum conversion?